### DISCUSSION TITLE
Looking for: LateX packages compatible with Avalonia Version >11.0.0 (& plotting packages)

### DISCUSSION THREAD

**Discussion by NoizeDaemon** [AvaloniaUI/Avalonia#11457] [Q&A] [answered]
Hi!

### I'm looking for a way to display math formulae in Avalonia, preferably using LateX syntax (since I'm already familiar with it), as well as packages for plotting graphs.

I tried `CSharpMath.Avalonia`, `AvaloniaMath` for LateX, and `ScottPlot` for plotting so far, but they all seem incompatible with newest previews giving errors like:
```
System.IO.FileNotFoundException: 'Could not load file or assembly 'Avalonia.Visuals, Version=0.10.19.0, Culture=neutral, PublicKeyToken=c8d484a7012f9a8b'
```

Since I want to target `Web` as well, I can't downgrade.

`LiveChartsCore.SkiaSharpView.Avalonia` works fine for plotting, however it does currently not offer basic functionality that I would need (being able to freely positions axes - since in math plotting axes are often centered & the goal would be to provide students with visuals closest to what they would be required to plot by hand).

### Can anyone suggest alternatives for math/formulae rendering and plotting that I could try or that are known to be compatible?

Here's a sample image of what kind of plots and math text I would like to achieve:

![image](https://github.com/AvaloniaUI/Avalonia/assets/42150677/d6217a65-44c8-4cc8-b632-38041e4ec03d)

I'm aware of this list: https://github.com/AvaloniaCommunity/awesome-avalonia#charts--plots--diagrams 
I'm gonna try `OxyPlot` next.

### UPDATE:

As for plotting, I got LiveCharts to somewhat work for me:

![image](https://github.com/AvaloniaUI/Avalonia/assets/42150677/895fc66f-2ab8-4173-9300-d651c967b62a)

(It's not pretty code-wise.. left has points added to create the arrows - which looks good, but obviously behaves badly when zooming/panning.. right has two line series added for the axes, and a bunch of points for the arrows & indicators.. so it behaves even worse, but oh well.. it's good enough for a static coordinate system 🤷‍♀️)

As for LaTex, I'm gonna try to update the CSharpMath lib - we'll see how that goes 😅

**Comment by timunie** [comment, marked as answer]
@NoizeDaemon in this situation, I'd clone the source of the mentioned libs and check if there is a preview branch already. If not, I'd try to update the lib on my own and get it running. Add it as a project reference instead of a nuget reference. 

Side-note: maybe the maintainer would be happy about a PR for the needed updates, if you want to do this.

**Reply by NoizeDaemon** [reply]
Thank you for the suggestion - I'm not very experienced, but I can give it a shot! :)
